erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -147,6 +148,12 @@ def validate_fixed_asset(self):
 					elif asset.status in ("Scrapped", "Cancelled", "Sold"):
 						frappe.throw(_("Row #{0}: Asset {1} cannot be submitted, it is already {2}").format(d.idx, d.asset, asset.status))
 
+	def validate_item_cost_centers(self):
+		for item in self.get("items",None):
+			cc = frappe.get_doc("Cost Center",item.cost_center)
```

Comment:
get_doc is expensive and you are calling it inside for loop. Instead fetch company of cost_center using `frappe.get_cached_value`, as in most cases cost center will be same for all items.